src/components/App/App.jsx

```diff
@@ -0,0 +1,283 @@
+/*   PLEASE READ THE README FILE */
+
+import { Route, Routes, useNavigate, useLocation } from "react-router-dom";
+import { useState, useEffect } from "react";
+import "./App.css";
+import Header from "../Header/Header";
+import Main from "../Main/Main";
+import Footer from "../Footer/Footer";
+import RegisterModal from "../RegisterModal/RegisterModal";
+import LoginModal from "../LoginModal/LoginModal";
+import EditProfileModal from "../EditProfileModal/EditProfileModal";
+import RecipeModal from "../RecipeModal/RecipeModal";
+import SearchedRecipes from "../SearchedRecipes/SearchedRecipes";
+import Profile from "../Profile/Profile";
+import ProtectedRoute from "../ProtectedRoute/ProtectedRoute";
+import About from "../About/About";
+import CurrentUserContext from "../../contexts/CurrentUserContext";
+import { getToken, setToken } from "../../utils/token";
+import {
+  registerUser,
+  signInUser,
+  updateUser,
+  isValidToken,
+} from "../../utils/auth";
+import { deleteRecipeCard, saveRecipe, getRecipeItems } from "../../utils/api";
+
+function App() {
+  const [activeModal, setActiveModal] = useState("");
+  //const [recipes, setRecipes] = useState([]);
+  const [savedRecipes, setSavedRecipes] = useState([]);
+  const [recommended, setRecommended] = useState([]);
+  const [currentUser, setCurrentUser] = useState(null);
+  const [isLoggedIn, setIsloggedIn] = useState(false);
+  const [selectedRecipe, setSelectedRecipe] = useState(null);
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const handleRegisterUser = () => {
+    setActiveModal("register");
+  };
+
+  const handleLoginUser = () => {
+    setActiveModal("login");
+  };
+
+  const handleEditProfileUser = () => {
+    setActiveModal("edit-profile");
+  };
+
+  const handleRecipeSummaryOpen = (recipe) => {
+    setActiveModal("summary");
+    setSelectedRecipe(recipe);
+    console.log(recipe);
+  };
+
+  const closeActiveModal = () => {
+    setActiveModal("");
+  };
+
```

Comment:
**COULD BE IMPROVED**

You can add an `Escape` listener like this to be able to remove it **correctly** when modals are closed:

```js
   useEffect(() => {

    if (!activeModal) return; // stop the effect not to add the listener if there is no active modal

    const handleEscClose = (e) => {  // define the function inside useEffect not to lose the reference on rerendering
      if (e.key === "Escape") {
        handleCloseModal();
      }
    };

    document.addEventListener("keydown", handleEscClose);

    return () => {  // don't forget to add a clean up function for removing the listener
      document.removeEventListener("keydown", handleEscClose);
    };
  }, [activeModal]);  // watch activeModal here
```

Here you define ` handleEscClose` inside the `useEffect` so that it could maintain a **fixed** reference if the component rerenders.
And of course we need to return a `clean up` function to **`remove`** the listener if the component unmounts.
And don’t forget about the **`dependencies array`** in order to watch `activeModal`